src/context/example.rs

```diff
@@ -0,0 +1,53 @@
+use report::example::ExampleReport;
+
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub enum ExampleLabel {
+    It,
+    Example,
+    Then,
+}
+
+impl From<ExampleLabel> for &'static str {
+    fn from(label: ExampleLabel) -> Self {
+        match label {
+            ExampleLabel::It => "It",
+            ExampleLabel::Example => "Example",
+            ExampleLabel::Then => "Then",
+        }
+    }
+}
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct ExampleInfo {
+    pub label: ExampleLabel,
+    pub name: String,
+    pub failure: Option<String>,
+}
+
+pub struct Example<T> {
+    pub(crate) info: ExampleInfo,
```

Comment:
Depends on whether we want to change `Event` from `pub` to `pub(crate)`.
Right now one can write one's own custom `EventHandler`.
The `…Info`, and thus all other contained types structs are part of `Event`.